test/e2e/test-document-height.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describes.endtoend(
+  'documentHeight',
+  {
+    testUrl:
+      'http://localhost:8000/test/fixtures/e2e/amp-carousel/0.1/document-height.html',
+    environments: ['viewer-demo'],
+  },
+  async (env) => {
+    it('should send documentHeight once amp has completed init', async () => {
+      const messages = env.receivedMessages;
+      const documentHeights = messages.filter(
+        (msg) => msg[0] === 'documentHeight'
+      );
+      const firstHeight = documentHeights[0][1].height;
```

Comment:
Nit: A little confusing since "height" can mean either a message or a number. You can also consider using an object instead of an subarray.
```js
const [type, height] = heightMessages[0];
```

Reply:
In the above suggestion, i'd need to write `height.height` in the expectation which is just as confusing!  I'm not sure why the `documentHeight` message sends an object instead of a plain number. I think I'm missing something about your suggestion re. using an object, unless I make a much bigger change / change the viewer api I'm not sure if I can change that.

I've renamed `documentHeights` to `documentHeightMessages` which should make it an ounce less confusing

Reply:
Oops, should use `[type, data]` then. 

> I think I'm missing something about your suggestion re. using an object

I meant how we construct `receivedMessages` in viewer.html (optional).

Reply:
Ah. I like that :). Done!